xds/src/test/java/io/grpc/xds/XdsSdsClientServerTest.java

```diff
@@ -88,8 +88,11 @@
   private TlsContextManagerImpl tlsContextManagerForServer;
 
   @Before
-  public void setUp() throws IOException {
+  public void setUp() throws Exception {
     port = XdsServerTestHelper.findFreePort();
+    URI expectedUri = new URI("sdstest://localhost:" + port);
+    fakeNameResolverFactory = new FakeNameResolverFactory.Builder(expectedUri).build();
+    NameResolverRegistry.getDefaultRegistry().register(fakeNameResolverFactory);
```

Comment:
Is this change required because of the change to the Registry or is this just an improvement to this test?

Reply:
oh this is to fix a bug that the test case registered `fakeNameResolverFactory` twice. Moving `register` to `@Before` guarantees it registered only once. Otherwise the leaked provider will cause the next test fail.

Reply:
oh do you mean [this](https://github.com/grpc/grpc-java/pull/8323/files#diff-cb9cb5fa575a01d40895ffc9883bc1026b65a40fb426f311f26f47f01a3e63b6L98)?, it's already like that. The previous error comes from picking the provider that didn't deregister previously thus failing the test code verification, it has nothing to do with the registry component.

Reply:
The link doesn't take me anywhere useful. But I do see there is a deregister within the `@After`. So that means getStub() is called twice. Seems that is only done in `mtlsClientServer_changeServerContext_expectException()`. Okay, this seems fine.